### PR TITLE
chore(core,express,fastify): fix hard dependencies with security issues

### LINKED ISSUE #13955: path-to-regexp npm audit high vulnerable
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

In express, @nestjs/core, @nestjs/platform-express there is a package used "path-to-regexp" what causes a npm high security vulnerable. For Version 0.1.7 there is a path 0.1.10 but for 3.2.0 there is no patch currently available. 
This should be updated. 

### Minimum reproduction code

https://github.com/pillarjs/path-to-regexp/commit/29b96b4a1de52824e1ca0f49a701183cc4ed476f

### Steps to reproduce

npm install
npm audit

### Expected behavior

no high security vulnerable

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [X] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [X] <code>@nestjs/platform-express</code>
- [X] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### Other package

_No response_

### NestJS version

10.3.10

### Packages versions

latest

### Node.js version

20

### In which operating systems have you tested?

- [X] macOS
- [X] Windows
- [X] Linux

### Other

_No response_

### LINKED ISSUE #13959: body-parser npm audit high vulnerability
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

In @nestjs/platform-express, there is the package "body-parser" in version 1.20.2 (https://github.com/nestjs/nest/blob/master/packages/platform-express/package.json#L21) what causes a npm high security vulnerability. There is a patch in body-parser version 1.20.3. Express has already updated this library: https://github.com/expressjs/express/blob/master/package.json#L33 

This should be updated.

### Minimum reproduction code

https://github.com/advisories/GHSA-qwcr-r2fm-qrc7

### Steps to reproduce

1. npm install
2. npm audit

### Expected behavior

No high security vulnerability

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [X] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### Other package

_No response_

### NestJS version

10.4.1

### Packages versions

[System Information]
OS Version     : Linux 5.15.153.1-microsoft-standard-WSL2
NodeJS Version : v20.17.0
NPM Version    : 10.8.3

[Nest CLI]
Nest CLI Version : 10.4.5

[Nest Platform Information]
platform-express version : 10.4.1
cache-manager version    : 2.2.2
schematics version       : 10.1.4
throttler version        : 6.2.1
mongoose version         : 10.0.10
terminus version         : 10.2.3
swagger version          : 7.4.0
testing version          : 10.4.1
common version           : 10.4.1
config version           : 3.2.3
axios version            : 3.0.3
core version             : 10.4.1
jwt version              : 10.2.0
cli version              : 10.4.5

### Node.js version

20.17.0

### In which operating systems have you tested?

- [ ] macOS
- [ ] Windows
- [X] Linux

### Other

_No response_

### LINKED ISSUE #13960: Vulnerability in Express < 4.20.0
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

Express is used by @nestjs/platform-express in version 4.19.2: https://github.com/nestjs/nest/blob/master/packages/platform-express/package.json#L23

### Minimum reproduction code

https://github.com/expressjs/express/security/advisories/GHSA-qw6h-vgh9-j6wx

### Steps to reproduce

See https://github.com/expressjs/express/security/advisories/GHSA-qw6h-vgh9-j6wx

### Expected behavior

Update to Express 4.20.0, which has been released on September 10 2024: https://github.com/expressjs/express/releases/tag/4.20.0

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [X] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### Other package

_No response_

### NestJS version

10.4.1

### Packages versions

[System Information]
OS Version : Linux 5.15.153.1-microsoft-standard-WSL2
NodeJS Version : v20.17.0
NPM Version : 10.8.3

[Nest CLI]
Nest CLI Version : 10.4.5

[Nest Platform Information]
platform-express version : 10.4.1
cache-manager version : 2.2.2
schematics version : 10.1.4
throttler version : 6.2.1
mongoose version : 10.0.10
terminus version : 10.2.3
swagger version : 7.4.0
testing version : 10.4.1
common version : 10.4.1
config version : 3.2.3
axios version : 3.0.3
core version : 10.4.1
jwt version : 10.2.0
cli version : 10.4.5

### Node.js version

20.17.0

### In which operating systems have you tested?

- [ ] macOS
- [ ] Windows
- [X] Linux

### Other

_No response_

### PR DESCRIPTION
**Pull request by micalevisk**
@kamilmysliwiec 

I just merged the changes made on PRs #13963 #13961 #13964

To address the security issues at `@nestjs/core`, `@nestjs/platform-express` and `@nestjs/platform-fastify` packages without introducing any breaking change

- closes #13959
- closes #13960 
- closes #13955

- closes #13965

### CODE REVIEW

**Review by Tony133 — APPROVED** [review]

**Review by brunoglm — APPROVED** [review]

**Review by rolandsusans — APPROVED** [review]

**Comment by sandorduba** [comment]
Hi, @kamilmysliwiec I'm new to open-source libraries and need this fix in our project. When can I expect it to be released?

File: packages/platform-fastify/package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "@fastify/middie": "8.3.1",
```
Comment (lrocheWB) [inline packages/platform-fastify/package.json:23]: this version also contains a vulnerability you should upgrade to version 8.3.3

Comment (micalevisk) [reply packages/platform-fastify/package.json:23]: is there any link where we can see that? because there were no reports on `npm audit` 

image

Comment (lrocheWB) [reply packages/platform-fastify/package.json:23]: image

https://github.com/fastify/middie/releases/tag/v8.3.3

**Pull request merged**